### PR TITLE
Basic fold

### PR DESCRIPTION
**Pull request by ptpham**
Only the mountain fold without geometry at the moment.

### CODE REVIEW

File: sheet.js [hunk 1]
```diff
@@ -1,41 +1,71 @@
...
 
   while (visiting.length) {
-    visit = visiting.shift();
-    if (!this.isOutside(visit.mirror)) {
-      visiting.push(visit.mirror);
-    }
-    if (!visited.has(visit.next)) {
-      visiting.push(visit.next);
-    }
-    visited.add(visit);
```
Comment (ptpham) [inline sheet.js]: I'm just being a bit militant about style here ... T__T

File: ordering.js [hunk 2]
```diff
@@ -62,28 +62,32 @@ Ordering.prototype.remove = function(elem) {
...
-
-  _.each(elems, function(elem) { _this.remove(elem); });
+  var result = this.impose(elems);
+  for (elem of elems) this.remove(elem); 
   this.rebind();
+  return result;
+};
 
+Ordering.prototype.impose = function(elems) {
+  var array = []; for (elem of elems) array.push(elem);
```
Comment (ptpham) [inline ordering.js]: Node doesn't have an Array.from or list comprehension in both ES5 and ES6 modes. T___T

Comment (ptpham) [inline ordering.js]: And lodash doesn't work properly with sets ...

File: test/sheet.js [hunk 3]
```diff
@@ -5,19 +5,26 @@ var assert = require('assert'),
...
        c  d.m <- b
   */
-  var cycle = HalfEdge.cycle(),
-      a = cycle.split(),
-      b = a.split(),
-      c = b.split(),
-      d = cycle.connect(b);
+  var sheet = new Sheet(),
+      base = sheet.sentinel.mirror,
+      a = sheet.split(base),
```
Comment (cixzhang) [inline test/sheet.js]: May want to test split since we're using it for the test.

File: test/sheet.js [hunk 4]
```diff
@@ -5,19 +5,26 @@ var assert = require('assert'),
 
 describe('sheet', function () {
   /**
-    cycle > d   a
+    base > d   a
```
Comment (cixzhang) [inline test/sheet.js]: Add a dash before the ```>``` to make up for the extra space? :D

File: ordering.js (hunk 2 above)
Comment (cixzhang) [inline ordering.js]: I wonder if it might make sense to extend loads with some set specific functions.

File: sheet.js [hunk 5]
```diff
@@ -1,41 +1,71 @@
...
+Sheet.prototype.connect = function(left, right) {
+  var result = left.connect(right);
+  this.ordering.merge(left, result);
+  this.ordering.merge(right, result.mirror);
+  return result;
+};
+
+Sheet.prototype.seamCompare = function(seam) {
+  var mirror = _.chain(seam).pluck('mirror').min(this.orderIteree).value();
+  var primary = _.min(seam, this.orderIteree, this);
```
Comment (cixzhang) [inline sheet.js]: ```this.orderIteree``` is already bound. Though if we want to do it like this every time, we could move it to prototype.

File: test/sheet.js [hunk 6]
```diff
@@ -5,19 +5,26 @@ var assert = require('assert'),
...
       |  ,  /    v
        c  d.m <- b
   */
-  var cycle = HalfEdge.cycle(),
-      a = cycle.split(),
-      b = a.split(),
-      c = b.split(),
-      d = cycle.connect(b);
+  var sheet = new Sheet(),
+      base = sheet.sentinel.mirror,
```
Comment (cixzhang) [inline test/sheet.js]: This is a bit strange to me. Doesn't seem that obvious that when you want to reference the first half edge in the sheet, you'd need to get the mirror of a sentinel.

File: sheet.js [hunk 7]
```diff
@@ -1,41 +1,71 @@
...
-  this.inside = new Set([this.halfedge]);
+function Sheet() {
+  var halfedge = HalfEdge.cycle();
+  this.sentinel = halfedge.mirror;
+  this.outside = new Set([halfedge.mirror]);
+  this.inside = new Set([halfedge]);
+  this.ordering = new Ordering();
+  this.ordering.merge(halfedge, halfedge.mirror);
+  this.orderIteree = (function(edge) {
+    return this.ordering.resolve(edge).index; }).bind(this);
```
Comment (cixzhang) [inline sheet.js]: Would a function like this make more sense in Ordering?

File: ordering.js (hunk 2 above)
Comment (ptpham) [inline ordering.js]: I've updated my node and it actually has Array.from! =D

File: test/sheet.js (hunk 3 above)
Comment (ptpham) [inline test/sheet.js]: I did add some tests below that test whether the split properly assigns an ordering -- this is all split does at this level. The lower level functionality should be tested in halfedge.js

File: test/sheet.js (hunk 4 above)
Comment (ptpham) [inline test/sheet.js]: Addressed.

File: sheet.js (hunk 5 above)
Comment (ptpham) [inline sheet.js]: Addressed. Also fixed spelling to iteratee.

File: test/sheet.js [hunk 8]
```diff
@@ -5,19 +5,26 @@ var assert = require('assert'),
 
 describe('sheet', function () {
   /**
-    cycle > d   a
+    initial -> d   a
```
Comment (cixzhang) [inline test/sheet.js]: Well, now that the word is longer, could we reduce the spacing a bit? Sorry xD

**Pull request merged**